Source/WebCore/css/parser/CSSSelectorParser.cpp

```diff
@@ -882,9 +890,13 @@ std::unique_ptr<CSSParserSelector> CSSSelectorParser::consumePseudo(CSSParserTok
         case CSSSelector::PseudoClassType::Any: {
             SetForScope resistDefaultNamespace(m_resistDefaultNamespace, true);
             auto selectorList = makeUnique<CSSSelectorList>();
-            *selectorList = consumeComplexForgivingSelectorList(block);
+            auto consumedBlock = consumeComplexForgivingSelectorList(block);
             if (!block.atEnd())
                 return nullptr;
+            if (consumedBlock.isEmpty())
+                *selectorList = CSSSelectorList { };
+            else
+                *selectorList = CSSSelectorList { WTFMove(consumedBlock) };
```

Comment:
CSSSelectorList constructor could probably handle this case.

Reply:
There is a ASSERT_SECURITY so I wasn't too sure about removing

https://github.com/WebKit/WebKit/blob/main/Source/WebCore/css/CSSSelectorList.cpp#L49